### PR TITLE
Fix Python Plugin DLL path issues when EnergyPlus is called through a symlink

### LINKED ISSUE #8037: Running EnergyPlus v9.3 through symlink produces DLL error
Issue overview
--------------
There are PATH related issues on unix systems when using a symlink to execute EnergyPlus:

```
**FATAL:DLL problem cause program termination
libc++abi.dylib: terminating with uncaught exception of type EnergyPlus::FatalError: DLL problem cause program termination
Abort trap: 6
```

This is related to needing to find the executable directory for the new API.

### Details
Some additional details for this issue (if relevant):
 - MacOS (symlink installed only through DMG installer)
 - EnergyPlus v9.3

### PR DESCRIPTION
**Pull request by nealkruis**
Pull request overview
---------------------
 - Fixes #8037
 
Resolves symlink path for executable in Python Plugin code.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally

### CODE REVIEW

**Comment by nealkruis** [comment]
@Myoldmopar I've confirmed this with local testing after setting LINK_WITH_PYTHON=ON.

@matthew-larson is adding a unit test file for FileSystem functions in #8008. I'll likely add a unit test after that is merged to test that we can get absolute paths for symlinks on all platforms (except Windows where we aren't creating any symlinks--and where I'm not sure there's a reliable way to support them).

File: src/EnergyPlus/FileSystem.cc [hunk 1]
```diff
@@ -106,6 +106,13 @@ namespace FileSystem {
 
     std::string getAbsolutePath(std::string const &path)
     {
+        /*
```
Comment (Myoldmopar) [inline src/EnergyPlus/FileSystem.cc:109]: :+1: I've never liked a comment-heavy PR so much before

File: src/EnergyPlus/PluginManager.cc [hunk 2]
```diff
@@ -127,7 +127,7 @@ namespace PluginManagement {
         // In this case, the current executable path will be the E+ install root
         // Which is where we find the Python Wrapper Shared Library
         if (wrapperDLLHandle) return; // already found
-        std::string programPath = FileSystem::getProgramPath();
+        std::string programPath = FileSystem::getAbsolutePath(FileSystem::getProgramPath());
```
Comment (Myoldmopar) [inline src/EnergyPlus/PluginManager.cc:130]: Yeah, this is awesome.  Thanks for the quick fix!

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
This is marked as draft, but looks good to me.  Unmark and I think this can go in.  

I added a new [test](https://github.com/Myoldmopar/EPTravisTester/pull/12/files#diff-396e1a6ba12bee989db994da6f1238a4R21-R26) for symlink runs to my sanity tester.  Once this PR goes in, I can kick off a package build and have Travis test it out on all the platforms.  Currently it [fails](https://travis-ci.com/github/Myoldmopar/EPTravisTester/jobs/344642809#L115) on Mac.

**Comment by Myoldmopar** [comment]
> This is marked as draft, but looks good to me. Unmark and I think this can go in.

Let me know if you are planning more here.

**Comment by nealkruis** [comment]
The only other thing I can imagine is adding a unit test for symlink resolution in FileSystem. If you don't think that's necessary, I think it can merge.

**Comment by Myoldmopar** [comment]
My thought is that we will be going to C++17 which will allow us to use the built-in FileSystem library.  At which point, we won't have our own file system code that we have to unit test anymore.  I believe that will happen soon, so I don't think it's worth sweating that for right now.  I'm going to merge this fix in.

**Pull request merged**